src/views/RecIM/components/Forms/EditMatchStatsForm/index.js

```diff
@@ -0,0 +1,245 @@
+      menuItems: matchStatus.map((type) => {
+        return type.Description;
+      }),
+      error: errorStatus.Name,
+      helperText: '*Required',
+    },
+  ];
+
+  const allFields = [createMatchStatsField].flat();
+
+  const currentInfo = useMemo(() => {
+    return {
+      TeamID: teamMatchHistory.TeamID,
+      Score: teamMatchHistory.TeamScore,
+      Sportsmanship: '',
+      StatusID: teamMatchHistory.Status ?? '',
+    };
+  }, [teamMatchHistory]);
+
+  const [newInfo, setNewInfo] = useState(currentInfo);
+  const [isSaving, setSaving] = useState(false);
+  const [openConfirmWindow, setOpenConfirmWindow] = useState(false);
+  const [disableUpdateButton, setDisableUpdateButton] = useState(true);
+
+  const handleSetError = (field, condition) => {
+    const getCurrentErrorStatus = (currentValue) => {
+      return {
+        ...currentValue,
+        [field]: condition,
+      };
+    };
+    setErrorStatus(getCurrentErrorStatus);
+  };
+
+  // refresh dropdown select once fetch is complete
+  useEffect(() => {
+    setNewInfo(currentInfo);
+  }, [currentInfo]);
+
+  // Field Validation
+  useEffect(() => {
+    let hasError = false;
+    let hasChanges = false;
+    for (const field in currentInfo) {
+      if (currentInfo[field] !== newInfo[field]) {
+        hasChanges = true;
+      }
+      handleSetError(field, newInfo[field] === '');
+      hasError = newInfo[field] === '' || hasError;
+    }
+    setDisableUpdateButton(hasError || !hasChanges);
+  }, [newInfo, currentInfo]);
+
+  const handleChange = (event, src) => {
+    const getNewInfo = (currentValue) => {
+      // datetime pickers return value rather than event,
+      // so we can also manually specify target source and value
+      if (src) {
+        let newValue = event;
+        return {
+          ...currentValue,
+          [src]: newValue,
+        };
+      }
+      return {
+        ...currentValue,
+        [event.target.name]:
+          event.target.type === 'checkbox' ? event.target.checked : event.target.value,
+      };
+    };
+    setNewInfo(getNewInfo);
+  };
+
+  const getFieldLabel = (fieldName) => {
+    const matchingField = allFields.find((field) => field.name === fieldName);
+    return matchingField.label;
+  };
+
+  function getNewFields(currentInfo, newInfo) {
+    const updatedFields = [];
+    Object.entries(newInfo).forEach(([key, value]) => {
+      if (currentInfo[key] !== value)
+        updatedFields.push({
+          Field: key,
+          Value: value,
+          Label: getFieldLabel(key),
+        });
+    });
+    return updatedFields;
+  }
+
+  const handleConfirm = () => {
+    debugger;
+    setSaving(true);
+    let matchStatsRequest = { ...currentInfo, ...newInfo };
+    matchStatsRequest.Score = parseInt(matchStatsRequest.Score);
+    matchStatsRequest.Sportsmanship = parseInt(matchStatsRequest.Sportsmanship);
+    matchStatsRequest.TeamID = parseInt(matchStatsRequest.TeamID);
+    matchStatsRequest.StatusID = parseInt(
```

Comment:
we don't need to parse int if we handle field validation (which was set to errorStatus.Name which did not exist)

Reply:
I removed this, but am waiting for patch to be fixed before I can see if the typescript can handle the change.

Reply:
It seems that the TS is ok with having them be passed as strings, so I will resolve this for now.